Bio/Align/fasta_m8.py

```diff
@@ -0,0 +1,238 @@
+                fields = line[len(prefix) :].strip().split(", ")
+                assert fields[0] == "query id"
+                assert fields[1] == "subject id"
+                assert fields[2] == "% identity"
+                assert fields[3] == "alignment length"
+                assert fields[4] == "mismatches"
+                assert fields[5] == "gap opens"
+                assert fields[6] == "q. start"
+                assert fields[7] == "q. end"
+                assert fields[8] == "s. start"
+                assert fields[9] == "s. end"
+                assert fields[10] == "evalue"
+                assert fields[11] == "bit score"
+                if fields[12] == "BTOP":
+                    self._alignment_representation = "BTOP"
+                elif fields[12] == "aln_code":
+                    self._alignment_representation = "CIGAR"
+                else:
+                    raise ValueError("Unexpected field '%s'" % fields[12])
+                line = next(stream)
+                line = line.strip()
+                assert line.startswith("# ")
+                suffix = " hits found"
+                assert line.endswith(suffix)
+                hits = int(line[2 : -len(suffix)])
+            else:
+                yield self.create_alignment(line)
+
+    def create_alignment(self, line):
+        """Parse one line of FASTA output and return an Alignment object."""
+        columns = line.split()
+        assert len(columns) == 13
+        annotations = {}
+        annotations["program"] = self._program
+        annotations["database"] = self._database
+        if self._query_id is not None:
+            assert columns[0] == self._query_id
+        query_id = columns[0]
+        target_id = columns[1]
+        percentage_identity = float(columns[2])
+        alignment_length = int(columns[3])
+        mismatches = int(columns[4])
+        matches = alignment_length - mismatches
+        difference = abs(100 * matches / alignment_length - percentage_identity)
+        assert difference < 0.015
+        gap_opens = int(columns[5])
+        query_start = int(columns[6]) - 1
+        query_end = int(columns[7])
+        target_start = int(columns[8]) - 1
+        target_end = int(columns[9])
+        annotations["mismatches"] = mismatches
+        annotations["evalue"] = float(columns[10])
+        annotations["bit_score"] = float(columns[11])
+        if self._alignment_representation == "BTOP":
+            coordinates = self.parse_btop(columns[12])
+        elif self._alignment_representation == "CIGAR":
+            coordinates = self.parse_cigar(columns[12])
+        coordinates[0, :] += target_start
+        if query_start < query_end:
+            coordinates[1, :] += query_start
+        else:
+            # mapped to reverse strand
+            coordinates[1, :] = query_start - coordinates[1, :] + 1
+        if self._query_size is None:
+            query_size = query_end
```

Comment:
This looks like an important corner case to have test coverage - do you have a test file which triggered it?

Reply:
With the current code, this cannot actually happen. I will remove it.